rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -380,31 +380,21 @@ impl<'me, K: ArrowReadableKey<'me> + Into<KeyWrapper>, V: ArrowReadableValue<'me
         self.load_blocks(&target_block_ids).await;
     }
 
-    pub(crate) async fn get(&'me self, prefix: &str, key: K) -> Result<V, Box<dyn ChromaError>> {
+    pub(crate) async fn get(
+        &'me self,
+        prefix: &str,
+        key: K,
+    ) -> Result<Option<V>, Box<dyn ChromaError>> {
         let search_key = CompositeKey::new(prefix.to_string(), key.clone());
         let target_block_id = self.root.sparse_index.get_target_block_id(&search_key);
         let block = self.get_block(target_block_id).await;
-        let res = match block {
-            Ok(Some(block)) => block.get(prefix, key.clone()),
+        match block {
+            Ok(Some(block)) => Ok(block.get(prefix, key.clone())),
             Ok(None) => {
                 tracing::error!("Block with id {:?} not found", target_block_id);
-                return Err(Box::new(ArrowBlockfileError::BlockNotFound));
```

Comment:
I assume we are deprecating `BlockNotFound`. Can we remove it from the corresponding error enum?

Reply:
BlockNotFound is still in use within `rust/blockstore/src/arrow/blockfile.rs`